### ISSUE TITLE
WSOD on filter of a property

### ISSUE THREAD

**Issue by amitaibu** [RESTful-Drupal/restful#340]
``` php
    $public_fields['user'] = array(
      'property' => 'author',
      'resource' => array(
        'user' => array(
          'name' => 'users',
        ),
      ),
    );
```

`https://example.com/api/foo?filter[user]=1`

``` javascript
{
type: "http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.5.1",
title: "SQLSTATE[42S22]: Column not found: 1054 Unknown column 'node.author' in 'where clause'",
status: 500
}
```

The reason is that in \RestfulDataProviderEFQ::queryForListFilter we should get the real property from the one defined in `$public_fields` which is in fact the $wrapper property.

In short: PR will follow :)

**Comment by e0ipso** [comment]
Shouldn't this be

```
https://example.com/api/foo?filter[user]=1
```

?

**Comment by amitaibu** [comment]
Sorry, yes it's `https://example.com/api/foo?filter[user]=1`. Still, without #341 - you would get an error ;)

**Comment by amitaibu** [comment]
Closed in favor of #341
